app/components/image/imageController.js

```diff
@@ -49,6 +54,24 @@ function ($scope, $stateParams, $state, Image, ImageHelper, Messages) {
     });
   };
 
+  $scope.pullImage = function(tag) {
+    var items = tag.split(":");
+    var image = items[0];
+    tag = items[1];
+    $('#loadingViewSpinner').show();
+    Image.create({fromImage: image, tag: tag}, function (d) {
```

Comment:
Instead of using `Image.create` directly, please use `ImageService.pullImage`. It returns a promise and can be handled with `.then()` and `.catch()` to manage success/errors.

See https://github.com/portainer/portainer/blob/develop/app/components/endpointAccess/endpointAccessController.js#L42 for an example of how to handle the promise.